pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/FunctionApiResource.java

```diff
@@ -54,7 +54,7 @@ public String clientAppId() {
                 : null;
     }
 
-    public AuthenticationDataHttps clientAuthData() {
-        return (AuthenticationDataHttps) httpRequest.getAttribute(AuthenticationFilter.AuthenticatedDataAttributeName);
+    public AuthenticationDataSource clientAuthData() {
```

Comment:
`AuthenticationDataHttps` is the return value of the error, use `AuthenticationDataSource` instead of it